crates/sui-core/src/checkpoints/mod.rs

```diff
@@ -343,48 +413,39 @@ impl CheckpointBuilder {
             )
         } else {
             current_gas_costs
-        };
-        let summary = CheckpointSummary::new(
-            self.state.epoch(),
-            sequence_number,
-            &contents,
-            previous_digest,
-            epoch_rolling_gas_cost_summary,
-            None, //todo
-        );
-
-        self.output
-            .checkpoint_created(&summary, &contents, last_checkpoint_of_epoch)
-            .await?;
-
-        self.metrics
-            .transactions_included_in_checkpoint
-            .inc_by(contents.size() as u64);
-        self.metrics
-            .last_constructed_checkpoint
-            .set(sequence_number as i64);
-
-        batch = batch.insert_batch(
-            &self.tables.checkpoint_content,
-            [(contents.digest(), contents)],
-        )?;
-        batch = batch.insert_batch(
-            &self.tables.checkpoint_summary,
-            [(sequence_number, summary)],
-        )?;
-        for txn in l.iter() {
-            batch = batch.insert_batch(
-                &self.tables.digest_to_checkpoint,
-                [(txn.transaction_digest, sequence_number)],
-            )?;
         }
+    }
 
-        Ok(batch)
+    async fn augment_epoch_last_checkpoint(
+        &self,
+        epoch_total_gas_cost: &GasCostSummary,
+        effects: &mut Vec<TransactionEffects>,
+    ) -> anyhow::Result<()> {
+        // TODO: We need to find a way to be able to cancel the advance epoch tx cert cosntruction
+        // process, ideally when reconfiguration is about to happen from outside.
```

Comment:
Under what scenario would the reconfiguration happen from outside? Is it when we want to manually trigger epoch changes for some reason? If so would it suffice to have a flag in CheckpointBuilder that's settable from outside, and if the flag is set we are in manual mode and we ignore the `last_checkpoint_of_epoch` flag.

Reply:
There is a separate checkpoint sync process, that may trigger reconfiguration when it synced the last checkpoint of the epoch. When that happens I need to make sure we can stop this process.